src/app/CommandHandler.h

```diff
@@ -411,31 +406,23 @@ class CommandHandler : public Messaging::ExchangeDelegate
     Access::SubjectDescriptor GetSubjectDescriptor() const
     {
         VerifyOrDie(!mGoneAsync);
-        return mExchangeCtx->GetSessionHandle()->GetSubjectDescriptor();
+        return mDispatcher.GetSubjectDescriptor();
     }
 
 private:
     friend class TestCommandInteraction;
     friend class CommandHandler::Handle;
-
-    CHIP_ERROR OnMessageReceived(Messaging::ExchangeContext * ec, const PayloadHeader & payloadHeader,
-                                 System::PacketBufferHandle && payload) override;
-
-    void OnResponseTimeout(Messaging::ExchangeContext * ec) override
-    {
-        //
-        // We're not expecting responses to any messages we send out on this EC.
-        //
-        VerifyOrDie(false);
-    }
+    // TODO would be much better if we provided a Closure instead of using friend here.
+    // This is so Dispatcher can call close.
+    friend class CommandHandlerDispatcher;
 
     enum class State : uint8_t
     {
         Idle,                ///< Default state that the object starts out in, where no work has commenced
         Preparing,           ///< We are prepaing the command or status header.
         AddingCommand,       ///< In the process of adding a command.
         AddedCommand,        ///< A command has been completely encoded and is awaiting transmission.
-        CommandSent,         ///< The command has been sent successfully.
+        DispatchingResponse, ///< The command response are being dispatched.
```

Comment:
Singular or plural?  State is singular, comment is confused about which it is...